src/libpst.c

```diff
@@ -3564,8 +3622,13 @@ static int pst_getBlockOffsetPointer(pst_file *pf, pst_id2_tree *i2_head, pst_su
         }
     }
     else {
+        DEBUG_WARN(("Found internal %#x value.\n", offset));
         // internal index reference
         size_t subindex  = offset >> 16;
+        if (pf->do_read64 == 2) {
+            // Shift over 3 more bits for new flags.
+            subindex = subindex >> 3;
```

Comment:
Ok, thanks for the explanation! I didn't realize it predated the spec (and by 8 years!)... which is probably the reason for a lot of the bugs we've found.